src/Illuminate/Auth/Access/Gate.php

```diff
@@ -39,35 +39,35 @@ class Gate implements GateContract
     /**
      * All of the defined abilities.
      *
-     * @var array
+     * @var array<string, callable>
      */
     protected $abilities = [];
 
     /**
      * All of the defined policies.
      *
-     * @var array
+     * @var array<class-string<\Illuminate\Database\Eloquent\Model>, class-string>
      */
     protected $policies = [];
 
     /**
      * All of the registered before callbacks.
      *
-     * @var array
+     * @var callable[]
      */
     protected $beforeCallbacks = [];
 
     /**
      * All of the registered after callbacks.
      *
-     * @var array
+     * @var callable[]
```

Comment:
Just like with the `string` typing, you can also narrow the `callable` one further down:
```suggestion
     * @var (callable(\Illuminate\Contracts\Auth\Authenticatable, string, bool|null, array): bool|null)[]
```

Reply:
Because of this line: https://github.com/imanghafoori1/laravel_framework/blob/gate-docblocks/src/Illuminate/Auth/Access/Gate.php#L584

Sure, one could argue that it might not be _intended_ to return `null`, but the code reads like it _could_ do so

Reply:
I mean the third parameter and not the return type.